XXX-url.md

```diff
@@ -0,0 +1,170 @@
+| Title  | Implement WHATWG URL Spec   |
+|--------|-----------------------------|
+| Author | @jasnell                    |
+| Status | DRAFT                       |
+| Date   | 2016-06-30T09:00:00-07:00   |
+
+## Description
+
+The WHATWG URL Standard specifies updated syntax, parsing and serialization of
+URLs as currently implemented by the main Web Browsers. The existing Node.js
+`url` module parsing and serialization implementation currently does not support
+the URL standard and fails to pass 160 of the WHATWG URL parsing tests.
+
+This proposal is to implement the WHATWG URL Standard by modifying the existing
+`url` module to provide an implementation of the `URL` object and associated
+APIs. Doing so improves the robustness of URL parsing, provides consistency
+with browser js code, and can eventually allow the reduction of node.js specific
+APIs.
+
+Initially, the implementation would be introduced as an undocumented
+experimental feature exposed via a new `URL` property in the `url` module.
```

Comment:
I'm all for it being a global.
On Jul 2, 2016 10:18 AM, "Chris Dickinson" <notifications@github.com> wrote:

> In XXX-url.md
> <https://github.com/nodejs/node-eps/pull/28#discussion_r69380482>:
>
> > +
> > +## Description
> > +
> > +The WHATWG URL Standard specifies updated syntax, parsing and serialization of
> > +URLs as currently implemented by the main Web Browsers. The existing Node.js
> > +`url` module parsing and serialization implementation currently does not support
> > +the URL standard and fails to pass 160 of the WHATWG URL parsing tests.
> > +
> > +This proposal is to implement the WHATWG URL Standard by modifying the existing
> > +`url` module to provide an implementation of the `URL` object and associated
> > +APIs. Doing so improves the robustness of URL parsing, provides consistency
> > +with browser js code, and can eventually allow the reduction of node.js specific
> > +APIs.
> > +
> > +Initially, the implementation would be introduced as an undocumented
> > +experimental feature exposed via a new `URL` property in the `url` module.
>
> There's already a Url property that points to the old constructor/parser
> — this is likely to cause confusion. *If we go down the route of
> supporting a new URL parser*, we may as well expose it as a global. Code
> written for the browser expects it to be a global, we may as well avoid
> artificially increasing the delta between browser JS and server JS.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node-eps/pull/28/files/a0a892210e9fa20f684247dd58e838a9cd333df1#r69380482>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAa2ecg3l3xtEecYClWoO40LgnxvIjUEks5qRp1pgaJpZM4IrtaC>
> .
>